tests/product/timing_test_decorator.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import sys
+
+from time import time
+
+
+def _initialize_module_logger(module_name):
```

Comment:
So I suggest we remove this method completely and make this piece of code execute at the module level. In the functions where you refer to the `logger` variable make sure you include the line `global logger`. Also, I suggest we hardcode the module to "root". I think that's fine because the reason why you would want to have different loggers for every module is to be able to adjust the logging level and logging destination at a more granular level whereas we just print everything at INFO to the console.

I did some experiments and determined that if the code is moved to be module level it will only get executed once.